sdk/aqueduct/artifacts/none_artifact.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import uuid
+from typing import Any, Dict, Optional
+
+from aqueduct.artifacts import utils as artifact_utils
+from aqueduct.artifacts.artifact import Artifact
+from aqueduct.dag import DAG
+from aqueduct.enums import ArtifactType
+from aqueduct.error import AqueductError
+
+
+class NoneArtifact(Artifact):
+    """This class represents a generic artifact within the flow's DAG.
+    Currently, a generic artifact can be any artifact other than table, numeric, bool, or parameter.
```

Comment:
This entire comment is irrelevant to NoneArtifact right.

Reply:
I'm curious if we need a separate NoneArtifact, or can we do everything in GenericArtifact? @cw75 I know you use content == None, but I wonder if we could just allow for content to be Optional and set other internal fields. Conceptually, this artifact seems unnecessary to me.

Reply:
I discussed this with Cg on last friday and we decided to proceed with creating a separate NoneArtifact mainly because GenericArtifact is a temporary solution for artifact types that we did not get to come up with meaning type-specific functions eg( head() for table artifact). I think in the future, Cg wants to slowly pull artifact type out of the GenericAritifact type and assign them type specific functions. From my personal standpoint, NoneArtifact and GenericArtifact shows the same code implementation so it was kinda redundant.

Reply:
As long as we can distinguish cases where the content is expected to be None v.s. the content is None because we haven't evaluated the content yet (constructing the artifact class from a flow run), I'm ok with merging.

Reply:
I think we can, since we can use the `from_flow_run` flag to check the second case.